### PR TITLE
PassengerCards updates

### PR DESCRIPTION
**Pull request by jvaclavik**
- [x] Circled button in `PassengerCards` for Android and normal one for iOS 
- [x] Removable button for adding passengers
- [x] Showing `VisaInfo` in `PassengerDetail`
- [x] Create basic elements in `PassengerDetailScreen`
- [x] Create reusable wrapper for `MenuItem`

### CODE REVIEW

File: packages/universal-components/src/MenuGroup/MenuItem.js [hunk 1]
```diff
@@ -34,6 +33,7 @@ export default function MenuItem({
   labelStyle,
   hideActionIcon = false,
 }: Props) {
+  const iconRendered = icon != null;
   const menuIcon =
     icon != null
```
Comment (jvaclavik) [inline packages/universal-components/src/MenuGroup/MenuItem.js]: I really don't like this, but if I use `iconRendered ? ...` Flow shows errors. Do you have idea how not to repeat the same condition?

File: apps/core/scenes/resultDetail/ResultDetail.js [hunk 2]
```diff
@@ -98,6 +102,7 @@ class ResultDetail extends React.Component<Props, State> {
             passengerCards={passengerCards}
             onActionPress={this.handlePassengerEditPress}
             actionIconName="edit"
+            onPassengerAddPress={this.handleAddPassenger}
```
[the remark this answers is not in the document]
Comment (jvaclavik) [reply apps/core/scenes/resultDetail/ResultDetail.js]: Good idea, done

File: packages/universal-components/src/MenuGroup/MenuItemWrapper.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+type Props = {|
+  +children: React.Node,
+  +iconRendered?: boolean,
+|};
+
+export default function MenuItemWrapper({ children, iconRendered }: Props) {
+  return (
+    <View>
+      <View style={styles.wrapper}>{children}</View>
+      <Separator style={iconRendered ? styles.noIcon : styles.iconSeparator} />
```
Comment (JoseDuSV) [inline packages/universal-components/src/MenuGroup/MenuItemWrapper.js]: Separator margin still seems to be slightly off on iOS.

Screenshot 2019-03-11 at 14 25 39

File: packages/components/src/passengerCards/PassengerCards.js [hunk 4]
```diff
@@ -12,10 +12,25 @@ import type {
...
 |};
 
+const AddPassengerIos = ({ onPress }) => (
+  <Button onPress={onPress} label="Add another passenger" block />
+);
+
+const AddPassengerAndroid = ({ onPress }) => (
+  <Button
+    onPress={onPress}
+    width={parseFloat(defaultTokens.heightButtonNormal)}
```
[the remark this answers is not in the document]
Comment (jvaclavik) [reply packages/components/src/passengerCards/PassengerCards.js]: OK, done

File: apps/core/scenes/passengerDetail/PassengerDetail.js [hunk 5]
```diff
@@ -1,8 +1,103 @@
...
+        </View>
+      </MenuGroup>
+      <MenuGroup title="Manage">
+        <MenuItem
+          icon={
+            // $FlowExpectedError clashing with class RNW$Text extends React$Component<RNW$Text$Props> from react-native-web flow-typed definitions
+            <ShareIcon />
+          }
+          title="Invite co-traveller"
+          onPress={this.onPressInvite}
```
[the remark this answers is not in the document]
Comment (jvaclavik) [reply apps/core/scenes/passengerDetail/PassengerDetail.js]: Fixed

File: packages/universal-components/src/MenuGroup/MenuItemWrapper.js (hunk 3 above)
Comment (jvaclavik) [reply packages/universal-components/src/MenuGroup/MenuItemWrapper.js]: Fixed

File: packages/components/src/passengerCards/PassengerCards.js [hunk 6]
```diff
@@ -29,27 +44,24 @@ export default class PassengerCards extends React.Component<Props> {
...
-              color={defaultTokens.colorTextButtonPrimary}
-            />
-          </Button>
-        </View>
+        {onAddPassengerPress && (
+          <View style={styles.addButtonContainer}>
+            {Platform.OS === 'ios' ? (
+              <AddPassengerIos onPress={onAddPassengerPress} />
+            ) : (
+              <AddPassengerAndroid onPress={onAddPassengerPress} />
```
[the remark this answers is not in the document]
Comment (jvaclavik) [reply packages/components/src/passengerCards/PassengerCards.js]: Done

File: packages/components/src/passengerCards/AddPassengerButton.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import { Icon, Button, StyleSheet } from '@kiwicom/universal-components';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+
+type Props = {|
+  +onPress: () => void,
+|};
+
+export default function AddPassengerButton({ onPress }: Props) {
+  return (
+    <View style={styles.addPassengerAndroid}>
```
Comment (JoseDuSV) [inline packages/components/src/passengerCards/AddPassengerButton.js]: I think this wrapper and its style is unnecessary.
It seems that it is required only to solve problem with oval button on Android. But we will probably need to fix this problem directly inside `Button` component (it seems that width prop currently only affects `Touchable` inside button which is probably not enough for Android version).

I suggest to remove this wrapper and create separate issue for oval circled button on Android.

File: packages/universal-components/src/MenuGroup/MenuItem.js (hunk 1 above)
Comment (JoseDuSV) [reply packages/universal-components/src/MenuGroup/MenuItem.js]: What about removing `iconRendered` completely. For example like below.
And then just check if any icon will be rendered.

Screenshot 2019-03-12 at 17 13 05

Comment (jvaclavik) [reply packages/universal-components/src/MenuGroup/MenuItem.js]: Done

File: packages/components/src/passengerCards/AddPassengerButton.js (hunk 7 above)
Comment (jvaclavik) [reply packages/components/src/passengerCards/AddPassengerButton.js]: Done (issue here: https://github.com/kiwicom/margarita/issues/528)

**Pull request merged**